src/vehicle.cpp

```diff
@@ -1666,77 +1761,110 @@ bool vehicle::split_vehicles( std::vector<std::vector <int>> new_vehs,
-            if( split_mounts.empty() ) {
-                // calculate effective mount point
-                point new_mnt = parts[ mov_part ].mount - mnt_offset;
-                new_vehicle->parts.back().mount = rotate_mount( face.dir(), new_vehicle->face.dir(),
-                                                  point( 0, 0 ), new_mnt );
-            } else {
-                new_vehicle->parts.back().mount = split_mounts[ new_part ];
+            if( !split_mounts.empty() ) {
+                new_mount = split_mounts[ new_part ];
+                new_vehicle->parts.back().mount = new_mount;
             }
-            // put the passenger on the new vehicle
-            if( passenger ) {
-                g->m.board_vehicle( passenger->pos(), passenger );
+            // remove labels associated with the mov_part
+            std::string label_str;
+            const auto iter = labels.find( label( cur_mount.x, cur_mount.y ) );
+            if( iter != labels.end() ) {
+                label_str = iter->text;
+                labels.erase( iter );
+                new_labels.insert( label( new_mount.x, new_mount.y, label_str ) );
             }
-#if 0
-            // add the label to the new vehicle
-            if( !label_str.empty() ) {
-                // vpart_position comes from where? WTF.
+            // remove the passenger from the old new vehicle
+            if( passenger ) {
+                parts[ mov_part ].remove_flag( vehicle_part::passenger_flag );
+                parts[ mov_part ].passenger_id = 0;
             }
-#endif
             // indicate the part needs to be removed from the old vehicle
             parts[ mov_part].removed = true;
             removed_part_count++;
         }
         g->m.dirty_vehicle_list.insert( new_vehicle );
         g->m.set_transparency_cache_dirty( smz );
-        new_vehicle->refresh();
-        new_vehicle->shift_if_needed();
+        if( !new_labels.empty() ) {
+            new_vehicle->labels = new_labels;
+        }
+
+        if( !split_mounts.empty() ) {
+            // include refresh
+            new_vehicle->shift_parts( point( 0, 0 ) - mnt_offset );
+        } else {
+            new_vehicle->refresh();
+        }
+
+        // update the precalc points
+        new_vehicle->precalc_mounts( 1,
+                                     new_vehicle->skidding ? new_vehicle->turn_dir : new_vehicle->face.dir(),
+                                     new_vehicle->pivot_point() );
+        if( !passengers.empty() ) {
+            new_vehicle->relocate_passengers( passengers );
+        }
     }
     return did_split;
 }
 
+bool vehicle::split_vehicles( std::vector<std::vector <int>> new_vehs )
+{
+    std::vector<vehicle *> null_vehicles;
+    std::vector<std::vector <point>> null_mounts;
+    std::vector<point> nothing;
+    null_vehicles.assign( new_vehs.size(), nullptr );
+    null_mounts.assign( new_vehs.size(), nothing );
```

Comment:
`std::vector` has a constructor that does this (it takes a count and an object, just like `assign` and does the same).

Reply:
Fixed.